### PR TITLE
crimson/osd: guard non-pg-op handling with with_sequencer()

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/osd/osd_operations/client_request.cc [hunk 1]
```diff
@@ -87,11 +87,12 @@ seastar::future<> ClientRequest::start()
 	return interruptor::with_interruption([this, pgref]() mutable {
           epoch_t same_interval_since = pgref->get_interval_start_epoch();
           logger().debug("{} same_interval_since: {}", *this, same_interval_since);
-          return with_sequencer(
-            interruptor::wrap_function([this, pgref]() -> interruptible_future<> {
+          auto interruptible_do_op =
```
Comment (rzarzynski) [inline src/crimson/osd/osd_operations/client_request.cc]: nit: changing `return with_sequencer` to `auto interruptible_do_op = ...` belongs to the previous commit, I think.

File: src/crimson/osd/osd_operations/client_request.cc [hunk 2]
```diff
@@ -108,16 +108,18 @@ seastar::future<> ClientRequest::start()
...
+                        process_pg_op(pgref) :
+                        process_op(pgref));
               });
-            })
-          ).then_interruptible([pgref]() {
-            return seastar::stop_iteration::yes;
-          });
+            });
+          // keep the ordering of non-pg ops when across pg internvals
+          return (is_pg_op() ?
```
Comment (rzarzynski) [inline src/crimson/osd/osd_operations/client_request.cc:117]: `is_pg_op()` iterates over the op vector, IIRC. How about doing it once, at  beginning and capturing the value in the lambda?

Comment (tchaikov) [reply src/crimson/osd/osd_operations/client_request.cc]: sure. updated and repushed.

File: src/crimson/osd/osd_operations/client_request.cc (hunk 1 above)
Comment (tchaikov) [reply src/crimson/osd/osd_operations/client_request.cc]: fixed.

**Review by rzarzynski — APPROVED** [review]

**Pull request merged**